src/block_neural.rs

```diff
@@ -188,86 +188,45 @@ pub fn new_neuron_block(
-            OptimizerData::<L> {
-                optimizer_data: self.optimizer.initial_data()
-            };
-            self.weights_len as usize
-        ];
-        self.rng_scratchpad = vec![0; self.num_neurons];
-        // We need to seed each layer with a separate seed... how?
-        // by the time we call this function input_offset and output_offset are set and are unique. L
-        self.rng = Xoshiro256PlusPlus::seed_from_u64(
-            (self.input_offset * self.output_offset + self.num_inputs + self.weights_len as usize)
-                as u64,
-        );
-
-        match self.init_type {
-            InitType::Xavier => {
-                let bound = 6.0_f64.sqrt() / ((self.num_inputs + self.num_neurons) as f64).sqrt();
-                let normal = Uniform::new(-bound, bound);
-                for i in 0..self.num_neurons * self.num_inputs {
-                    self.weights[i as usize].weight = normal.sample(&mut self.rng) as f32;
-                }
-            }
-            InitType::Hu => {
-                let normal =
-                    Normal::new(0.0, (2.0 / self.num_inputs as f64).sqrt() as f64).unwrap();
-                for i in 0..self.num_neurons * self.num_inputs {
-                    self.weights[i as usize].weight = normal.sample(&mut self.rng) as f32;
-                }
-            }
-            //            InitType::RandomFirst1 => { for i in 0..self.num_inputs { self.weights[i as usize].weight = 1.0}},
-            //            InitType::RandomFirst10 => { for i in 0..self.num_inputs { self.weights[i as usize].weight = 0.0}; self.weights[0].weight = 1.0;},
-            InitType::One => {
-                for i in 0..self.weights_len {
-                    self.weights[i as usize].weight = 1.0
-                }
-            }
-            InitType::Zero => {
-                for i in 0..self.weights_len {
-                    self.weights[i as usize].weight = 0.0
-                }
-            }
-        }
-
-        // Bias terms are always initialized to zero
-        for i in 0..self.num_neurons {
-            self.weights[(self.num_neurons * self.num_inputs + i) as usize].weight = 0.0
-        }
-    }
-
-    fn get_num_output_slots(&self) -> usize {
-        1
-    }
+impl<L: OptimizerTrait + 'static>  BlockNeuronLayer<L> {
+    #[inline(always)]
+    fn internal_forward(
+        &self,
+        pb: &mut port_buffer::PortBuffer,
+        alpha: f32
+    ) {
+        unsafe {
+            let (input_tape, output_tape) = block_helpers::get_input_output_borrows(
+                &mut pb.tape,
+                self.input_offset,
+                self.num_inputs,
+                self.output_offset,
+                self.num_neurons,
+            );
 
-    fn get_num_output_values(&self, output: graph::OutputSlot) -> usize {
-        assert!(output.get_output_index() == 0);
-        self.num_neurons
-    }
+            // This is actually speed things up considerably.
```

Comment:
Typo, if cleaning up, let's fix these too

Reply:
Removed the comment, doesn't add any context